### DISCUSSION TITLE
Is There Any Way to Update From HAA 3.0.2?

### DISCUSSION THREAD

**Discussion by albertshih3** [RavenSystem/esp-homekit-devices#1714] [Q&A]
I had two Tuya smart-life bulbs which I flashed with HAA (using Tuya-convert). One was flashed properly but the other bulb was accidentally flashed with an old version of HAA (version 3.0.2). As of now OTA updates don't work and since it is so old, there is not a lot of info about json configs and also HomeKit doesn't recognize the device. Is there any way to update the firmware? (I do NOT have any way to solder things).

If there is no way to do it that's okay!

**Comment by RavenSystem** [comment]
Has that old version custom server in setup web?

**Reply by albertshih3** [reply]
Yes it does

**Reply by RavenSystem** [reply]
I can not help you now because I am on iPhone. But you will can recover and update it.

**Reply by albertshih3** [reply]
Okay I will try and take a look at the custom server stuff myself for now. Thank you!
